tests/lib/ifaces/linux_bridge_iface_test.py

```diff
@@ -0,0 +1,420 @@
+        port_config[Port.VLAN_SUBTREE].pop(Vlan.TRUNK_TAGS)
+
+        iface = LinuxBridgeIface(iface_info)
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+    def test_validate_trunk_mixing_id_and_id_range(self):
+        iface_info = gen_bridge_iface_info_with_vlan_filter()
+        port_config = iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE][1]
+        port_config[Port.VLAN_SUBTREE][Vlan.TRUNK_TAGS][0].update(
+            TRUNK_TAGS_ID_RANGES[0]
+        )
+
+        iface = LinuxBridgeIface(iface_info)
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+    def test_validate_trunk_id_range_missing_min_or_max(self):
+        iface_info = gen_bridge_iface_info_with_vlan_filter()
+        port_config = iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE][1]
+        port_config[Port.VLAN_SUBTREE][Vlan.TRUNK_TAGS][0] = {
+            Vlan.TrunkTags.ID_RANGE: {Vlan.TrunkTags.MIN_RANGE: 400}
+        }
+
+        iface = LinuxBridgeIface(iface_info)
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+        port_config[Port.VLAN_SUBTREE][Vlan.TRUNK_TAGS][0] = {
+            Vlan.TrunkTags.ID_RANGE: {Vlan.TrunkTags.MAX_RANGE: 400}
+        }
+
+        iface = LinuxBridgeIface(iface_info)
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+    def test_validate_tag_in_access_mode_and_native_trunk(self):
+        iface_info = gen_bridge_iface_info_with_vlan_filter()
+        iface = LinuxBridgeIface(iface_info)
+
+        iface.pre_edit_validation_and_cleanup()
+
+    def test_validate_tag_in_non_native_trunk(self):
+        iface_info = gen_bridge_iface_info_with_vlan_filter()
+        port_conf = iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE][1]
+        port_conf[Port.VLAN_SUBTREE][Vlan.ENABLE_NATIVE] = False
+        iface = LinuxBridgeIface(iface_info)
+
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+    def test_validate_enable_native_in_access_mode(self):
+        iface_info = gen_bridge_iface_info_with_vlan_filter()
+        port_conf = iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE][0]
+        port_conf[Port.VLAN_SUBTREE][Vlan.ENABLE_NATIVE] = True
+        iface = LinuxBridgeIface(iface_info)
+
+        with pytest.raises(NmstateValueError):
+            iface.pre_edit_validation_and_cleanup()
+
+    def test_remove_slave(self):
+        iface_info = gen_bridge_iface_info()
+        expected_iface_info = gen_bridge_iface_info()
+        expected_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            deepcopy(SLAVE1_PORT_CONFIG)
+        ]
+
+        iface = LinuxBridgeIface(iface_info)
+        iface.remove_slave(SLAVE2_IFACE_NAME)
+
+        assert iface.to_dict() == expected_iface_info
+
+    def test_state_for_verify_set_vlan_subtree_explicitly(self):
```

Comment:
I don't understand this test; consider renaming it to hint at what it does.

Reply:
Renamed to `test_verify_when_vlantree_not_defined_in_desire()`.

Reply:
Thanks.